test/e2e/azure_machinepool_drain.go

```diff
@@ -157,13 +166,39 @@ func testMachinePoolCordonAndDrain(ctx context.Context, mgmtClusterProxy, worklo
 			return err
 		}
 
-		decreasedReplicas := *owningMachinePool.Spec.Replicas - int32(1)
+		decreasedReplicas = *owningMachinePool.Spec.Replicas - int32(1)
 		owningMachinePool.Spec.Replicas = &decreasedReplicas
 		return helper.Patch(ctx, owningMachinePool)
 	}, 3*time.Minute, 3*time.Second).Should(Succeed())
 
-	// TODO setup a watcher to validate expected 2nd order drain outcomes
-	// https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2159
+	By(fmt.Sprintf("checking for a machine to start draining for machine pool: %s/%s", amp.Namespace, amp.Name))
+	Eventually(func() error {
+		ampmls, err := getAzureMachinePoolMachines(ctx, mgmtClusterProxy, workloadClusterProxy, amp)
+		if err != nil {
+			LogWarning(errors.Wrap(err, "failed to list the azure machine pool machines").Error())
+			return err
+		}
+
+		for i := range ampmls {
+			if conditions.Has(&ampmls[i], clusterv1.DrainingSucceededCondition) && conditions.IsTrue(&ampmls[i], clusterv1.DrainingSucceededCondition) {
+				controllerutil.RemoveFinalizer(&ampmls[i], azureMachinePoolMachineTestFinalizer)
+				return nil // started draining the node prior to delete
+			}
+		}
+
+		return errors.New("no machine has started to drain")
+	}, 3*time.Minute, 3*time.Second).Should(Succeed())
```

Comment:
I wonder if 3 minutes isn't long enough for the 99.9 percentile outcome of a successful k8s drain operation

Reply:
So far, I haven't run into a single failure yet. Do drain operations typically take a longer time? If so it could be safer to increase this to maybe 5 minutes.

Reply:
I think 10 mins is probably an appropriate timeout val

Reply:
They are workload-dependent. If we're confident that the workloads we're running in our test environment can reliably shut down in a few seconds then we should be O.K.